Cython/Compiler/PyrexTypes.py

```diff
@@ -2253,6 +2344,14 @@ class CPySSizeTType(CIntType):
     def sign_and_name(self):
         return "Py_ssize_t"
 
+class CHPySSizeTType(CIntType):
```

Comment:
Interesting. Could you add a comment here on what makes this special?

Reply:
HPy defines its own size type, i.e. `HPy_ssize_t`, similar to `Py_ssize_t`. This is mainly done to avoid importing `Python.h` in HPy which would then mean that any HPy extension transitively depends on Python's C API.
I would say, there is nothing special about this type. I introduced it because it has a different name and because the conversion functions are also named differently.

Reply:
> HPy defines its own size type, i.e. `HPy_ssize_t`, similar to `Py_ssize_t`. This is mainly done to avoid importing `Python.h` in HPy which would then mean that any HPy extension transitively depends on Python's C API. I would say, there is nothing special about this type. I introduced it because it has a different name

This should be handled entirely on the C side then, not in the type system.

> and because the conversion functions are also named differently.

That sounds like a deliberate source of annoyance for users.